src/containerapp/azext_containerapp/_arc_utils.py

```diff
@@ -0,0 +1,358 @@
+        config_map = core_v1_api.read_namespaced_config_map(name=resource_name, namespace=resource_namespace, pretty=True, _preload_content=False)
+    except client.exceptions.ApiException as e:
+        if e.status == 404:
+            config_map = None
+        else:
+            raise e
+    except Exception as e:
+        raise ValidationError(f"other errors while getting config map coredns in kube-system {str(e)}")
+
+    return config_map
+
+
+def update_configmap(resource_name, resource_namespace, kube_client, config_map):
+    validate_resource_name_and_resource_namespace_not_empty(resource_name, resource_namespace)
+
+    try:
+        logger.info(f"Start to update configmap {resource_name} in namespace {resource_namespace}")
+        core_v1_api = client.api.core_v1_api.CoreV1Api(kube_client)
+        core_v1_api.patch_namespaced_config_map(name=resource_name, namespace=resource_namespace, body=config_map)
+
+    except Exception as e:
+        raise CLIError(f"other errors while patching config map coredns in kube-system {str(e)}")
+
+
+def replace_configmap(resource_name, resource_namespace, kube_client, config_map):
+    validate_resource_name_and_resource_namespace_not_empty(resource_name, resource_namespace)
+
+    try:
+        logger.info(f"Start to replace configmap {resource_name} in namespace {resource_namespace}")
+        core_v1_api = client.api.core_v1_api.CoreV1Api(kube_client)
+        core_v1_api.replace_namespaced_config_map(name=resource_name, namespace=resource_namespace, body=config_map)
+
+    except Exception as e:
+        raise CLIError(f"other errors while replacing config map coredns in kube-system {str(e)}")
+
+
+def delete_configmap(resource_name, resource_namespace, kube_client):
+    validate_resource_name_and_resource_namespace_not_empty(resource_name, resource_namespace)
+
+    try:
+        logger.info(f"Start to delete configmap {resource_name} in namespace {resource_namespace}")
+        core_v1_api = client.api.core_v1_api.CoreV1Api(kube_client)
+        core_v1_api.delete_namespaced_config_map(name=resource_name, namespace=resource_namespace)
+
+    except client.rest.ApiException as e:
+        if e.status == HTTPStatus.NOT_FOUND:
+            logger.info(f"Configmap {resource_name} in namespace {resource_namespace} has been deleted")
+            return
+        raise CLIError(f"other ApiException while deleting config map coredns in kube-system {str(e)}")
+
+    except Exception as e:
+        raise CLIError(f"other errors while deleting config map coredns in kube-system {str(e)}")
+
+
+def validate_resource_name_and_resource_namespace_not_empty(resource_name, resource_namespace):
+    if resource_name is None or len(resource_name) == 0:
+        raise InvalidArgumentValueError("Arg resource_name should not be None or Empty")
+    if resource_namespace is None or len(resource_namespace) == 0:
+        raise InvalidArgumentValueError("Arg namespace should not be None or Empty")
```

Comment:
It is better to specify the parameter which customers input incorrectly, like:
raise InvalidArgumentValueError("Arg --namespace should not be None or Empty")

Reply:
This parameter is not a cli command parameter, but an internal method parameter from other method. So I think I should not use --, but I can change it to resource_namespace to align with the parameter name.